packages/mobile/src/web3/saga.ts

```diff
@@ -143,15 +151,18 @@ export function* assignAccountFromPrivateKey(key: string) {
     }
 
     let account: string
-    if (isZeroSyncMode()) {
-      const privateKey = String(key)
+
+    // Save the account to a local file on the disk.
+    // This is done for all sync modes, to allow users to switch into zeroSync mode.
+    // Note that if geth is running it saves the key using web3.personal.
+    const privateKey = String(key)
+    account = getAccountAddressFromPrivateKey(privateKey)
+    yield savePrivateKeyToLocalDisk(account, privateKey, pincode)
```

Comment:
This makes me nervous. We allowed it for zerosync mode, which I half-assumed was just a temporary thing. but if we're now going to permanently store the user's private key (encrypted by their pincode which could be poor and/or lost), I think we should have a broader discussion about it first. Maybe we already have but I missed it?

Reply:
Thanks for flagging this. After looking into it, geth also stores the private keys in a file encrypted with the password (for which we use the same pincode), so I will keep this as is for now. 
See https://celo-org.slack.com/archives/CL7BVQPHB/p1571763622020000

Reply:
Ok